src/lib/parser.rs

```diff
@@ -225,24 +225,24 @@ impl Parser {
               continue;
             }
 
-            if let Some(must_match_surfix) = surfix_map.get(option) {
-              // This option contains a surfix match, lets test it here
-              let next_char = self.seek_next_char();
-              if let None = next_char {
-                continue;
-              } else if !must_match_surfix.contains(next_char.unwrap()) {
-                continue;
-              }
-            }
+            match meta_map.get(option) {
+              Some(meta) => {
+                if let Some(next_char_needs_to_match) = meta.after() {
+                  // This option contains a surfix match, test test it here
+                  let next_char = self.seek_next_char();
+                  if let None = next_char {
+                    continue;
+                  } else if !next_char_needs_to_match.contains(next_char.unwrap()) {
+                    continue;
+                  }
+                }
 
-            for opt in options {
-              if opt.0.into() == option {
-                return Some(opt.0);
+                return Some(meta)
               }
+              None => panic!("This should not happen, please create an issue with example code so we can resolve this"),
```

Comment:
I think that maybe we should provide a link to the repository rather than assume the user gets the context of the error message.

```suggestion
              None => panic!("An error has occured, please create an issue at https://github.com/talpalang/talpa/issues with your code so we can resolve this"),
```

Reply:
Sounds good!